root/static/scripts/edit/components/URLInputPopover.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+import ButtonPopover from '../../common/components/ButtonPopover';
+import type {ErrorTarget} from '../externalLinks';
+import {ERROR_TARGETS} from '../URLCleanup';
+
+type PropsT = {
+  errorMessage: string,
+  errorTarget: ErrorTarget,
+  onCancel: () => void,
+  onChange: (SyntheticEvent<HTMLInputElement>) => void,
+  onToggle: (boolean) => void,
+  rawUrl: string,
+  url: string,
+};
+
+const URLInputPopover = (props: PropsT): React.MixedElement => {
+  const popoverButtonRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const toggle = (open) => {
+    /*
+     * Will be called by ButtonPopover when closed
+     * either by losing focus or click 'Close' button
+     */
+    setIsOpen(open);
+    props.onToggle(open);
+  };
+
+  const handleCancel = () => {
+    props.onCancel();
+    toggle(false);
+  };
+
+  const buildPopoverChildren = (
+    closeAndReturnFocus,
+  ) => (
+    <form
+      onSubmit={(event) => {
+        event.preventDefault();
+        closeAndReturnFocus();
+      }}
+    >
+      <table>
+        <tbody>
+          <tr>
+            <td className="section">
+              {addColonText(l('URL'))}
+            </td>
+            <td>
+              <input
+                className="value raw-url"
+                onChange={props.onChange}
+                style={{width: '336px'}}
+                value={props.rawUrl}
+              />
+              {props.errorMessage &&
+                props.errorTarget === ERROR_TARGETS.URL &&
+                <div
+                  className="error field-error target-url"
+                  data-visible="1"
+                >
+                  {props.errorMessage}
+                </div>
+              }
+            </td>
+          </tr>
+          {props.url &&
+            <tr>
+              <td className="section" style={{whiteSpace: 'nowrap'}}>
+                {addColonText(l('Cleaned up to'))}
+              </td>
+              <td>
+                <a className="clean-url" href={props.url}>{props.url}</a>
```

Comment:
(This should have target="_blank" so people don't lose their editing progress by clicking on it.)
```suggestion
                <a className="clean-url" href={props.url} target="_blank">{props.url}</a>
```